src/utils.js

```diff
@@ -80,89 +75,36 @@ function returnNeighbouringLadCode(ladCode, searchLowerLadCode) {
+      val: get(dataByGeography).get(geoCode).get(category.code)["value"].toLocaleString(),
+      perc: get(dataByGeography).get(geoCode).get(category.code)["perc"],
+      unit: tables[category.table].unit,
+      geoCode: geoCode,
+    };
   }
 }
 
-export function calculateComparisonDiff(geoCode, comparatorGeoCode, totalCatCode, category) {
-  let comparatorTotal;
-  let comparatorVal;
-  if (comparatorGeoCode == config.eAndWGeoCode) {
-    comparatorTotal = get(englandAndWalesData).get(config.eAndWGeoCode).get(totalCatCode);
-    comparatorVal = get(englandAndWalesData).get(config.eAndWGeoCode).get(category.code);
-  } else {
-    comparatorTotal = get(dataByGeography).get(comparatorGeoCode).get(totalCatCode);
-    comparatorVal = get(dataByGeography).get(comparatorGeoCode).get(category.code);
+export function calculateComparisonDiff(geoCode, comparatorGeoCode, catCode) {
+  if (get(dataByGeography).has(geoCode) && get(dataByGeography).get(geoCode).has(catCode)) {
+    let comparatorCategory;
+    if (comparatorGeoCode == config.eAndWGeoCode) {
+      comparatorCategory = get(englandAndWalesData).get(config.eAndWGeoCode).get(catCode);
+    } else {
+      comparatorCategory = get(dataByGeography).get(comparatorGeoCode).get(catCode);
+    }
+    const localCategory = get(dataByGeography).get(geoCode).get(catCode);
+    const percentageDiff = ((localCategory.perc - comparatorCategory.perc) / comparatorCategory.perc) * 100;
+    return Math.round(percentageDiff * 10) / 10;
   }
-  const localTotal = get(dataByGeography).get(geoCode).get(totalCatCode);
-  const localVal = get(dataByGeography).get(geoCode).get(category.code);
-  const percentageDiff =
-    ((localVal / localTotal - comparatorVal / comparatorTotal) / (comparatorVal / comparatorTotal)) * 100;
-  return Math.round(percentageDiff * 10) / 10;
 }
 
-export const updateMapAndComparisons = (tableSlug, categorySlug, metadata, geoCode, neighbouringLad) => {
-  let category = getCategoryBySlug(tableSlug, categorySlug);
-  let table = category ? filterSelectedTable(metadata, category) : null;
-  let comparisons = {};
+export const updateMap = (category) => {
   if (category) {
-    fetchCensusData(new LegacyCensusDataService(), dbColumnToCategoryId(category.code), null);
-    fetchCensusDataBreaks(new MetadataApiDataService(), category.code, table.total.code, 5);
-  }
-  if (geoCode != config.eAndWGeoCode) {
-    if (get(dataByGeography).get(geoCode)) {
-      comparisons.eAndWDiff = calculateComparisonDiff(geoCode, config.eAndWGeoCode, table.total.code, category);
-    }
-    if (neighbouringLad && get(dataByGeography).get(neighbouringLad.code)) {
-      comparisons.neighbouringLadDiff = calculateComparisonDiff(
-        geoCode,
-        neighbouringLad.code,
-        table.total.code,
-        category,
-      );
-    }
+    fetchSelectedDataForGeoType(new GeodataApiDataService(), "lad", [category.code, tables[category.table].total]);
```

Comment:
should these things really be classes if they're just being instantiated to use one method, then dropped? Do they actually hold any state?